From 83eefaf0307f3df19c94355ef9dc4007b31e5072 Mon Sep 17 00:00:00 2001
From: Dong Chuanxiao <chuanxiao.dong@intel.com>
Date: Fri, 16 Aug 2013 17:28:35 +0800
Subject: [PATCH 242/429] mmc: host: add quirk to disable preset register

Due to Arason silicon issue, SDHCI hosts in BYT/MRFL cannot use preset
register. Add quirks for SDHCI hosts

Signed-off-by: Chuanxiao Dong <chuanxiao.dong@intel.com>
---
 drivers/mmc/host/sdhci-pci.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/mmc/host/sdhci-pci.c b/drivers/mmc/host/sdhci-pci.c
index 83344a17ee25..c4fff769da47 100644
--- a/drivers/mmc/host/sdhci-pci.c
+++ b/drivers/mmc/host/sdhci-pci.c
@@ -602,6 +602,7 @@ static int byt_sdio_probe_slot(struct sdhci_pci_slot *slot)
 }
 
 static const struct sdhci_pci_fixes sdhci_intel_byt_emmc = {
+	.quirks2	= SDHCI_QUIRK2_PRESET_VALUE_BROKEN,
 	.allow_runtime_pm = true,
 	.probe_slot	= byt_emmc_probe_slot,
 };
@@ -613,9 +614,7 @@ static const struct sdhci_pci_fixes sdhci_intel_byt_sdio = {
 };
 
 static const struct sdhci_pci_fixes sdhci_intel_byt_sd = {
-	.quirks2	= SDHCI_QUIRK2_PRESET_VALUE_BROKEN |
-		SDHCI_QUIRK2_POWER_PIN_GPIO_MODE,
-	.allow_runtime_pm = true,
+	.quirks2	= SDHCI_QUIRK2_PRESET_VALUE_BROKEN,
 	.probe_slot	= byt_sd_probe_slot,
 	.remove_slot	= byt_sd_remove_slot,
 };
-- 
2.37.3

